### PR TITLE
feat: trip vs shape distance validation

### LINKED ISSUE #163: Implement trip distance vs. shape verification (GTFS rule)
**Is your feature request related to a problem? Please describe.**
The distance traveled by a trip should be less or equal to the max length of its shape. This is a GTFS rule implemented in Google Python validator and featured in Google Type Error as TYPE_TRIP_WITH_OUT_OF_ORDER_SHAPE_DIST_TRAVELED.

**Describe the solution you'd like**
Actual Google GTFS validator behaviour : verifies if distance travelled is smaller than shape length. If not, generate an ERROR.

**Describe alternatives you've considered**

**Additional context**
Line 141 in Error support priorities https://docs.google.com/spreadsheets/d/1vqe6wq7ctqk1EhYkgtZ0_TbcQ91vccfs2daSjn20BLE/edit#gid=0

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

Created `trip_distance_exceeds_shape_distance`notice when the max value `shapeDistTraveled` of a trip is greater than the max value of `shapeDistTraveled`for its associated shape. 

Closes #163 

**Expected behavior:** 

Screenshot 2023-08-07 at 1 10 57 AM

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by cka-y** [comment]
@emmambd could you confirm what fields should be present in the notice's details?

**Comment by emmambd** [comment]
@cka-y List of fields is correct! This is enough to cross reference the error and find what the problem is.

**Comment by emmambd** [comment]
@cka-y re: rounding the decimal point, please cut off after three decimal points (0.000), since this is sufficient to tell the difference between the two `shape_dist_traveled` values in all the cases from the original analytics report. Don't round up or down, since the values should match what's in the .txt files.

**Review by qcdyx — APPROVED** [review]
LGTM

**Review by emmambd — APPROVED** [review]
After doing some spot checks and looking at the report, this looks good to me! Noting that the acceptance tests generate errors for 7% of feeds in the Mobility Data which is above the recommended threshold, but this rule does align with the community consensus from https://github.com/google/transit/pull/380/files. @cka-y One question, not a blocker: any obvious reason why the list of feeds in the acceptance report is slightly different from the one in the original Python analysis report?

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.*;
+
+@RunWith(JUnit4.class)
+public class TripAndShapeDistanceValidatorTest {
```
Comment (davidgamez) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidatorTest.java:15]: Great addition!

**Review by davidgamez — APPROVED** [review]
It's nicely done!

**Comment by cka-y** [comment]
@emmambd seems like the script was using the producer URL i.e. current version of the dataset as opposed to the action that uses a persisted version of the dataset. I'm going to keep an eye on this but I think it's safe to merge atm.

**Pull request merged**